### PR TITLE
[SDK][azure] Support specify compute instance as session compute

### PR DESCRIPTION
**Pull request by D-W-**
# Description
This pull request primarily introduces support for specifying a compute instance as a session compute in the `promptflow` package. It also includes several updates to the `AzureMachineLearningDesignerServiceClient` and its associated files, such as removing the `FlowRunsAdminOperations` operations and updating the AutoRest Code Generator version from 3.8.0 to 3.9.2.

Main changes:

* [`src/promptflow/CHANGELOG.md`](diffhunk://#diff-41ec3f7c4b5d4c0e670407d3c00a03a6966d7ebf617b1536473e33a12e2bc765R8): Added a new feature that supports specifying a compute instance as a session compute in `run.yaml`.
* [`src/promptflow/promptflow/_sdk/entities/_run.py`](diffhunk://#diff-14f4d7a1d9b077c7b749230a92628782f990e1fc7cb741b20ff894f868327020R551-R555): Added a new attribute `compute_name` to the `_to_rest_object` method to support the new feature. [[1]](diffhunk://#diff-14f4d7a1d9b077c7b749230a92628782f990e1fc7cb741b20ff894f868327020R551-R555) [[2]](diffhunk://#diff-14f4d7a1d9b077c7b749230a92628782f990e1fc7cb741b20ff894f868327020R579)
* [`src/promptflow/promptflow/_sdk/schemas/_run.py`](diffhunk://#diff-db38cc07d25efdbc0622e8e9352e07e34c502d36bdad1954bcd52db156192c9fR34-R35): Added a new field `compute` to the `ResourcesSchema` class.

Updates to the `AzureMachineLearningDesignerServiceClient` and its files:

* [`src/promptflow/promptflow/azure/_restclient/flow/_azure_machine_learning_designer_service_client.py`](diffhunk://#diff-0d897b1801e67d281d2e78b96e4735815629d7341c4ba5bb328326bd917719d9L3-R3): Removed `FlowRunsAdminOperations` operations from the `AzureMachineLearningDesignerServiceClient` class. Also, the AutoRest Code Generator version was updated. [[1]](diffhunk://#diff-0d897b1801e67d281d2e78b96e4735815629d7341c4ba5bb328326bd917719d9L3-R3) [[2]](diffhunk://#diff-0d897b1801e67d281d2e78b96e4735815629d7341c4ba5bb328326bd917719d9L15-L34) [[3]](diffhunk://#diff-0d897b1801e67d281d2e78b96e4735815629d7341c4ba5bb328326bd917719d9L44-L45) [[4]](diffhunk://#diff-0d897b1801e67d281d2e78b96e4735815629d7341c4ba5bb328326bd917719d9L83-R76)
* [`src/promptflow/promptflow/azure/_restclient/flow/aio/_azure_machine_learning_designer_service_client.py`](diffhunk://#diff-b86b4b5a24655b993946a74b33780deb5eeab5452feb269e74456ba3871de495L3-R3): Similar changes were made to the asynchronous version of the `AzureMachineLearningDesignerServiceClient` class. [[1]](diffhunk://#diff-b86b4b5a24655b993946a74b33780deb5eeab5452feb269e74456ba3871de495L3-R3) [[2]](diffhunk://#diff-b86b4b5a24655b993946a74b33780deb5eeab5452feb269e74456ba3871de495L16-R16) [[3]](diffhunk://#diff-b86b4b5a24655b993946a74b33780deb5eeab5452feb269e74456ba3871de495L39-L40) [[4]](diffhunk://#diff-b86b4b5a24655b993946a74b33780deb5eeab5452feb269e74456ba3871de495L61-R52)

Other changes:

* [`src/promptflow/promptflow/azure/_restclient/README.md`](diffhunk://#diff-cefa87b9a4d4ce5542bad1b2c2cad0f874ff643588255b5a16d6bc9bae59894cR22): Updated the changelog to reflect the new feature.
* Several other files in the `src/promptflow/promptflow/azure/_restclient/flow` directory were updated, mainly to reflect the new AutoRest Code Generator version and to make minor formatting changes.
Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/azure/_restclient/README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Download swagger.json from [here](https://int.api.azureml-test.ms/flow/swagger/v
 
 - 2023.11.13 - [Update SDK restclient](https://github.com/microsoft/promptflow/pull/1101).
 - 2023.12.18 - [Remove data portal url from the result of pfazure run show](https://github.com/microsoft/promptflow/pull/1497)
+- 2024.2.2 - [Support specify compute instance as session compute](https://github.com/microsoft/promptflow/pull/xx)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/azure/_restclient/README.md:22]: why pull/xx?

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/promptflow/azure/_restclient/README.md (hunk 1 above)
Comment (D-W-) [reply src/promptflow/promptflow/azure/_restclient/README.md]: Updated

File: src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py [hunk 2]
```diff
@@ -995,3 +995,51 @@ def test_eager_flow_download(self, pf: PFClient, simple_eager_run: Run):
...
+        run_id = randstr("run_id")
+        run = load_run(
+            source=source,
+            params_override=[{"name": run_id}],
+        )
+        rest_run = run._to_rest_object()
+        assert rest_run.compute_name == "my_ci"
+
+        # update ci to actual ci
+        run._resources["compute"] = compute_instance_name
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py:1040]: change this attribute in above `params_override` would also work?

**Review by 0mza987 — APPROVED** [review]

File: src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py (hunk 2 above)
Comment (D-W-) [reply src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py:1040]: Yes, I want to check the loaded YAML file without changing compute name also works, see line 1037.

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**